### PR TITLE
[TS] Implement support for SEP-7

### PR DESCRIPTION
**Pull request by CassioMG**
Ticket: https://stellarorg.atlassian.net/jira/software/c/projects/WAL/boards/37?selectedIssue=WAL-1449

This PR adds support for SEP-7 through the following classes:
- `Sep7Base`: abstract base class with common functions and attributes used by both `tx` and `pay` operations
- `Sep7Tx`: implements the [Sep-7 `tx` operation](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0007.md#operation-tx)
- `Sep7Pay`: implements the [Sep-7 `pay` operation](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0007.md#operation-pay)

This PR also adds some convenient parsing/helper functions as follow:
- `isValidSep7Uri`: checks if uri is Sep-7 compliant, if not returns a explanatory `reason` message
- `parseSep7Uri`: parses a Sep-7 uri to either a Sep7Tx or Sep7Pay instance
- `sep7ReplacementsFromString`: converts a complex replacements string to a list of friendly replacements objects
- `sep7ReplacementsToString`: converts a list of friendly replacements objects to a complex replacements string

Those classes and functions can be used to either `generate` a new Sep-7 URI or `handle` an existing Sep-7 URI.

This PR also adds a bunch of `tests` and `JSDocs` that should cover all new classes and functions.

This PR also adds a few missing types and exceptions from existing code (harmless changes).

### CODE REVIEW

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ * Returns true if the given URI is a SEP-7 compliant URI, false otherwise.
+ *
+ * Currently this checks whether it starts with 'web+stellar:tx' or 'web+stellar:pay'
+ * and has its required parameters: 'xdr=' and 'destination=' respectively.
+ *
+ * @param {string} uri The URI string to check.
+ *
+ * @returns {boolen} returns `true` if it's a valid Sep-7 uri, `false` otherwise.
+ */
+export const isValidSep7Uri = (uri: string): boolean => {
```
Comment (aristidesstaffieri) [inline @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:23]: Should this check that the `xdr` parameter for tx operations are a valid `TransactionEnvelope`?
Same question for `destination` on the pay operation, should it check that the value is a valid account ID?

**Review by aristidesstaffieri — APPROVED** [review]
left one question but lgtm otherwise

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts (hunk 1 above)
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:23]: Good call, I like your suggestions :)

I think I'll make this function returning something like:
```
{
  result: boolean;
  reason: string;
}
```

Where `reason` should be a message letting devs know why the verification failed.

What do you think?

Comment (aristidesstaffieri) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:23]: yeah that sounds good to me, I like the more informational response. 👍

Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts]: Implemented here: https://github.com/stellar/typescript-wallet-sdk/pull/141/commits/64f6544abef3a0e5c7e5a271f4d189f18b093415
More tests to account for the error reason messages: https://github.com/stellar/typescript-wallet-sdk/pull/141/commits/89c504566d2d5b61ab7f874113b44c78f8182b95

Comment (aristidesstaffieri) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts]: this lgtm 👏

**Pull request merged**